### PR TITLE
cleaned up api

### PR DESCRIPTION
**Pull request by moskalyk**

### CODE REVIEW

File: app/api/conversation/conversationModel.ts [hunk 1]
```diff
@@ -68,15 +74,16 @@ class Conversation {
   }
 
   //This will check that a conversation exists, if it does not, it wil return null
```
Comment (dharness) [inline app/api/conversation/conversationModel.ts]: comment still says null

Comment (dharness) [inline app/api/conversation/conversationModel.ts]: can we rename it to getConversation?

**Comment by moskalyk** [comment]
@dharness @nnur

File: app/api/conversation/conversationController.ts [hunk 2]
```diff
@@ -12,16 +12,16 @@ var convo = require('./conversationModel.js');
...
 
-router.get('/', async (req, res) => {
+router.get('/', async function(req, res) {
   return res.send(await convo.all());
 });
 
-router.get('/:fingerPrint', async (req, res) => {
-  return res.send(await convo.findById(req.params.fingerPrint));
+router.get('/:fingerPrint', async function(req, res) {
+  return res.send(await convo.findOrCreate(req.params.fingerPrint));
```
Comment (dharness) [inline app/api/conversation/conversationController.ts]: switch to post

File: app/api/conversation/conversationModel.ts [hunk 3]
```diff
@@ -17,9 +17,9 @@ class Conversation {
     var mostRecentMessageIds = []
 
     for (var i = 0; i < conversations.length; ++i)
-      mostRecentMessageIds.push(conversations[i].Messages[conversations[i].Messages.length - 1])
+      mostRecentMessageIds.push(conversations[i].Messages[conversations[i].Messages.length - 1]);
```
Comment (dharness) [inline app/api/conversation/conversationModel.ts]: can we pull `conversations[i].Messages[conversations[i].Messages.length - 1]` out into a variable?

File: app/api/conversation/conversationModel.ts [hunk 4]
```diff
@@ -39,24 +39,21 @@ class Conversation {
...
-  //Based on a unique fingerprint, 
-  //this will return the conversation with message
-  public async findOrCreate(AFingerprint, BFingerprint, roomId) {
-    
+  //Based on a unique roomId, 
+  //this will return the conversation with message, if not undefined
+  public async findOrCreate(AFingerprint: string, BFingerprint: string, roomId: string) {
     var conversation = await this.createConversation(AFingerprint, BFingerprint, roomId);
-
+    if(!conversation){
```
Comment (dharness) [inline app/api/conversation/conversationModel.ts]: Can we switch this to:
    var conversationData;
    
    if(conversation){
        conversationData = {
            messages: '..etc',
            conversation: conversation
        }
    }
    return conversationData

in the interest of having only one return statement?

**Comment by dharness** [comment]
LGTM after mentioned changes

**Pull request merged**